### ISSUE TITLE
issue with metadata for V2 model

### ISSUE THREAD

**Issue by imran-syed-abdul** [SAP-samples/hana-opensap-cloud-2020#28]
Hi Team,

I am getting error while retrieving metadata for V2 support. My server.js file 

 `const cds = require('@sap/cds')
const Sdk = require('@dynatrace/oneagent-sdk')
const DynaT = Sdk.createInstance()
const proxy = require('@sap/cds-odata-v2-adapter-proxy')
console.log(DynaT.getCurrentState())
console.log(`CDS Custom Boostrap from /srv/server.js`)
global.__base = __dirname + "/"
process.on('uncaughtException', function (err) {
    console.error(err.name + ': ' + err.message, err.stack.replace(/.*\n/, '\n')) // eslint-disable-line
})

const TextBundle = require("@sap/textbundle").TextBundle
// global.__bundle = new TextBundle("../_i18n/i18n", require("./utils/locale").getLocale())

cds.on('bootstrap', app => app.use(proxy()));
cds.on('served', () => {
    // add more middleware after all cds servies
})

// delegate to default server.js:
module.exports = async (o) => {
    o.port = process.env.PORT || 4001
    //API route (Disabled by default)
    // o.baseDir = process.cwd()
    o.baseDir = global.__base;    
    o.routes = []

    const express = require('express')
    let app = express()
    app.express = express
    // app.baseDir = process.cwd()
    app.baseDir = o.baseDir;
    o.app = app    
    const path = require('path')
    const fileExists = require('fs').existsSync
    let expressFile = path.join(app.baseDir, 'server/express.js')
    if (fileExists(expressFile)) {
        await require(expressFile)(app)
    }

    //V2 fallback
    cds.serve('POService')
        .from(global.__base + "/gen/csn.json")
        .to("fiori")
        .at('/POService').in(app)
        .catch((err) => {
            app.logger.error(err);
        })

    cds.serve('MasterDataService')
        .from(global.__base + "/gen/csn.json")
        .to("fiori")
        .at('/MasterDataService').in(app)
        .catch((err) => {
            app.logger.error(err);
        })

    o.app.httpServer = await cds.server(o)

    const glob = require('glob')
    //Load routes
    // let baseDir = process.cwd()
    let routesDir = path.join(global.__base, 'routes/**/*.js')
    let files = glob.sync(routesDir)
    this.routerFiles = files;
    if (files.length !== 0) {
        for (let file of files) {
            await require(file)(app, app.httpServer)
        }
    }

    return o.app.httpServer
}

// const cds = require("@sap/cds");
// const proxy = require("@sap/cds-odata-v2-adapter-proxy");

// cds.on("bootstrap", app => app.use(proxy()));

// module.exports = cds.server;`

**Comment by jung-thomas** [comment]
I'm a bit tied up with TechEd today and tomorrow, but I'll put this on my list to look into just as soon as TechEd is complete.

**Comment by jung-thomas** [comment]
I think I've found the issue.  Looks like there was some change in the recent version of update of CAP.  The fallback handler definitions in boostrap code are actually no longer needed. Instead in cds.on boostrap, you can map the OData V2 proxy paths to the correct services. I just updated this one section to the following:

`
cds.on('bootstrap', app => app.use(proxy({
    services: {
        "/MasterDataService/": "MasterDataService",
        "/POService/": "POService"
    }
})))
`
Then it works just fine.

![image](https://user-images.githubusercontent.com/12159356/101823299-8dc91c80-3af8-11eb-8d52-44833d91b7d8.png)

**Comment by imran-syed-abdul** [comment]
Hi Thomas,

Thanks a lot for checking, but not sure something is missing. Still facing the issue. Please find the code below.

[https://github.com/imran-syed-abdul/hana-opensap-cloud](url)

Error : GET /MasterDataService/$metadata
#2.0#2020 12 10 22:16:56:856#+00:00#ERROR#/cov2ap/MetadataRequest#####1ff79994-ea68-4322-b37d-f11987abd001##########1ff79994-ea68-4322-b37d-f11987abd001#PLAIN##request to http://localhost:4004/MasterDataService/$metadata failed, reason: connect ECONNREFUSED 127.0.0.1:4004#
Dec 10, 2020 10:16:56 PM Logger.js [1ff79994-ea68-4322-b37d-f11987abd001] ERROR: request to http://localhost:4004/MasterDataService/$metadata failed, reason: connect ECONNREFUSED 127.0.0.1:4004
FetchError: request to http://localhost:4004/MasterDataService/$metadata failed, reason: connect ECONNREFUSED 127.0.0.1:4004
    at ClientRequest.<anonymous> (/home/user/projects/hana-opensap-cloud-1/node_modules/node-fetch/lib/index.js:1461:11)
    at ClientRequest.emit (events.js:198:13)
    at Socket.socketErrorListener (_http_client.js:401:9)
    at Socket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:59:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)

**Comment by jung-thomas** [comment]
Ok I created a pull request to your repository with a fix:
https://github.com/imran-syed-abdul/hana-opensap-cloud/pull/1
To copy the explanation here:
Main thing I changed was removing /srv/routes/cds.js (by renaming the file). During the videos I move the boostraping server.js and the additional loading of the CDS service handlers in cds.js was no longer needed and conflicting. I also updated you package.json dependency to the latest version of @sap/cds. So be sure to run an npm install. I tested it all and it worked now.
